testutil-server/src/test/java/io/spine/testing/server/blackbox/BlackBoxBoundedContextTest.java

```diff
@@ -106,7 +112,66 @@ void tearDown() {
     T boundedContext() {
         return context;
     }
-    
+
+    @Test
+    @DisplayName("register command dispatchers")
+    void registerCommandDispatchers() {
+        CommandClass commandTypeToIntercept = CommandClass.from(BbRegisterCommandDispatcher.class);
+        BbCommandDispatcher dispatcher = new BbCommandDispatcher(context.eventBus(),
+                                                                 commandTypeToIntercept);
+        context.withHandlers(dispatcher);
+        context.receivesCommand(registerCommandDispatcher(dispatcher.getClass()));
+        assertThat(dispatcher.commandsIntercepted()).isEqualTo(1);
+    }
+
+    @Test
+    @DisplayName("throw on an attempt to register duplicate command dispatchers")
+    void throwOnDuplicateCommandDispatchers() {
+        CommandClass commandTypeToIntercept = CommandClass.from(BbRegisterCommandDispatcher.class);
+        BbCommandDispatcher dispatcher = new BbCommandDispatcher(context.eventBus(),
+                                                                 commandTypeToIntercept);
+        context.withHandlers(dispatcher);
+        assertThrows(IllegalArgumentException.class, () -> context.withHandlers(dispatcher));
```

Comment:
Instead, let's try to register a dispatcher of a different type which handles the same command.